### PR TITLE
Add projectile-discover-projects-in-directory

### PR DESCRIPTION
**Pull request by justbur**
This is a helper function to go through the subdirectories of a
directory, check which ones are valid projects and add them to the cache
file. I didn't make it recursive although it could be. 

-----------------

Before submitting a PR make sure the following things have been done (and denote this
by checking the relevant checkboxes):

- [x] The commits are consistent with our [contribution guidelines](../CONTRIBUTING.md)
- [ ] You've added tests (if possible) to cover your change(s)
- [ ] All tests are passing (`make test`)
- [x] The new code is not generating bytecode or `M-x checkdoc` warnings
- [x] You've updated the changelog (if adding/changing user-visible functionality)
- [x] You've updated the readme (if adding/changing user-visible functionality)

Thanks!

### CODE REVIEW

**Comment by justbur** [comment]
I triggered a new CI build because it looks like the last one failed because a server was down. 

By the way, a bunch of tests fail on windows, at least some of them can be fixed by using canonical paths, but I'm not sure if you care.

**Comment by bbatsov** [comment]
This should probably be added at least to the menu, otherwise no one is going to find out about it.

**Comment by bbatsov** [comment]
A mention in the manual would be useful as well.

**Comment by justbur** [comment]
@bbatsov ok done

File: projectile.el [hunk 1]
```diff
@@ -721,6 +721,22 @@ The cache is created both in memory and on the hard drive."
...
+at the top level of DIRECTORY."
+  (interactive
+   (list (read-directory-name "Starting directory: ")))
+  (let ((subdirs (directory-files directory t)))
+    (mapcar
+     (lambda (dir)
+       (when (and (file-directory-p dir)
+                  (not (member (file-name-nondirectory dir) '(".." "."))))
+         (let ((default-directory dir))
+           (projectile-cache-projects-find-file-hook))))
```
Comment (bbatsov) [inline projectile.el]: This is not correct. You should simply check whether you're in a project dir and if so add this dir to the list of known projects.

**Comment by justbur** [comment]
Thanks, please check it now

**Pull request merged**